src/modules/admin/admin.controller.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Controller,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  Get,
+  NotFoundException,
+} from '@nestjs/common';
+import { CreateUserDto } from './dtos/create-user.dto';
+import { UpdateUserDto } from './dtos/update-user.dto';
+import { UserEntity } from '../../database/entities/user.entity';
+import { AdminService } from './admin.service';
+
+@Controller('users')
+export class AdminController {
+  constructor(private readonly adminService: AdminService) {}
+
+  @Post()
+  async createUser(@Body() createUserDto: CreateUserDto) {
+    return this.adminService.createUser(createUserDto);
+  }
+
+  @Patch(':id')
+  async editUser(
+    @Param('id') id: string,
+    @Body() updateUserDto: UpdateUserDto,
+  ) {
+    return this.adminService.editUser(id, updateUserDto);
+  }
+
+  @Delete(':id')
+  async deleteUser(@Param('id') id: string) {
+    return this.adminService.deleteUser(id);
+  }
+
+  @Get()
+  async getAllUsers(): Promise<UserEntity[]> {
+    return this.adminService.getAllUsers();
+  }
+
+  @Get(':id')
+  async getUserById(@Param('id') id: string): Promise<UserEntity> {
+    const user = await this.adminService.getUserById(id);
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+    return user;
+  }
+
+  @Get(':email')
+  async getUserByEmail(@Param('email') email: string): Promise<UserEntity> {
+    const user = await this.adminService.getUserByEmail(email);
+    if (!user) {
+      throw new NotFoundException('User not found');
```

Comment:
não dá para personalizar essa mensagem?

Algo do tipo: "Email {x} was not found in user db"

Reply:
Vou deixar para os Dev em formação realizar esses detalhes simples, pois se eu realizar a atividade completa e com a mensagem de melhor agrado, vou ter realizado tudo sem meus colegas desenvolverem sua logica.